src/coreclr/vm/greenthreads.cpp

```diff
@@ -219,7 +228,10 @@ SuspendedGreenThread* GreenThread_StartThread(TakesOneParam functionToExecute, u
     GetThread()->SetActiveThreadBase(pGreenThread);
     pGreenThread->m_currentThreadObj = GetThread();
 
+    assert(t_greenThread.inGreenThread == false);
+    t_greenThread.greenThreadOnStack = true;
     GreenThread_StartThreadHelper((uintptr_t)FirstFrameInGreenThread, &detailsAboutWhatToCall);
+    t_greenThread.greenThreadOnStack = false;
```

Comment:
You tried but removed `assert(t_greenThread.inGreenThread == true);` here.  Is it always `false`?  Or is this a multiple exit conditions thing?

Reply:
Its a multiple exit conditions thing. if inGreenThread, we're exiting due to a suspension, else we just finished the green thread without ever suspending.